api/src/BcGov.Malt.Web/Services/LdapUserSearchService.cs

```diff
@@ -0,0 +1,156 @@
+            if (string.IsNullOrEmpty(configuration.Username)) missingSettings.Add(nameof(configuration.Username));
+            if (string.IsNullOrEmpty(configuration.Password)) missingSettings.Add(nameof(configuration.Password));
+
+            if (missingSettings.Count != 0)
+            {
+                _logger.LogError("Missing LDAP configuration settings {Settings}", missingSettings);
+
+                string joinedSettings = string.Join(", ", missingSettings.ToArray());
+                throw new ConfigurationErrorsException("Missing LDAP configuration settings: " + joinedSettings + ".");
+            }
+
+            return configuration;
+        }
+
+        private Task<SearchResultEntry> SendRequestAsync(LdapConnection connection, SearchRequest searchRequest)
+        {
+            // convert Begin/End to async
+            // example from https://github.com/dotnet/runtime/issues/28470
+            return Task.Factory.FromAsync((callback, state) =>
+            {
+                return connection.BeginSendRequest(searchRequest, PartialResultProcessing.ReturnPartialResultsAndNotifyCallback, callback, state);
+            },
+            (asyncresult) =>
+            {
+                SearchResponse searchResponse = (SearchResponse)connection.EndSendRequest(asyncresult);
+                if (searchResponse.ResultCode == ResultCode.Success && searchResponse.Entries.Count != 0)
+                {
+                    return searchResponse.Entries[0];
+                }
+
+                return null;
+            },
+            null);
+        }
+
+        private SearchRequest CreateSearchRequest(string distinguishedName, string username)
+        {
+            SearchRequest searchRequest = new SearchRequest();
+            searchRequest.DistinguishedName = distinguishedName;
+            searchRequest.Filter = $"(&(objectCategory=person)(objectClass=user)(sAMAccountName={username}))";
+
+            searchRequest.Attributes.Add("bcgovGUID");          // id
+            searchRequest.Attributes.Add("sAMAccountName");     // username
+            searchRequest.Attributes.Add("userPrincipalName");  // UPN
+            searchRequest.Attributes.Add("sn");                 // lastname
+            searchRequest.Attributes.Add("givenName");          // firstname
+            searchRequest.Attributes.Add("mail");               // email address
+
+            return searchRequest;
+        }
+
+        private User MapSearchResult(SearchResultEntry entry)
+        {
+            var user = new User
+            {
+                Id = GetAttributeValue(entry, "bcgovGUID") ?? string.Empty,
+                UserName = GetAttributeValue(entry, "sAMAccountName") ?? string.Empty,
+                FirstName = GetAttributeValue(entry, "givenName") ?? string.Empty,
+                LastName = GetAttributeValue(entry, "sn") ?? string.Empty,
+                Email = GetAttributeValue(entry, "mail") ?? string.Empty,
+            };

```

Comment:
We don't map UPN Here?

Reply:
UPN is not used by the front end.  I was going to address this when Nancy has the SharePoint stuff almost ready.

Reply:
Gotcha, was wondering if it was something along those lines. Just need to unblock merging then due to readme.md conflicts